usaspending_api/etl/management/commands/load_submission.py

```diff
@@ -647,14 +647,12 @@ def load_file_d1(submission_attributes, procurement_data, db_cursor):
         legal_entity_location, created = get_or_create_location(legal_entity_location_field_map, row, legal_entity_location_value_map)
 
         # Create the legal entity if it doesn't exist
-        try:
-            legal_entity = LegalEntity.objects.get(recipient_unique_id=row['awardee_or_recipient_uniqu'])
-        except ObjectDoesNotExist:
+        legal_entity, created = LegalEntity.get_or_create_by_duns(duns=row['awardee_or_recipient_uniqu'])
+        if created:
```

Comment:
I think at this point no. As much as I hate to suggest it we may just want to connect directly to SAMS and pull this information for ourselves rather than relying on it filtering in via submissions to the broker... :\